research/berry/berrylib/test_fast_math.py

```diff
@@ -0,0 +1,61 @@
+import jax
+import jax.numpy as jnp
+import numpy as np
+from berrylib.fast_math import (
+    jax_fast_invert,
+    jax_faster_inv,
+    jax_faster_inv_diag,
+    jax_faster_inv_product,
+    jax_faster_log_det,
+    log_normal_pdf,
+)
+
+
+def test_faster_linalg():
+    num_iter = 1000
+    key = jax.random.PRNGKey(0)
+    d = jax.random.uniform(key, (num_iter, 4))
+    s = jax.random.uniform(key, (num_iter, 1))
+    g = jax.random.uniform(key, (num_iter, 4))
+    atol = 1e-6
+    for d, s, g in zip(d, s, g):
+        m = np.diag(d) + s
+        sign, expected = np.linalg.slogdet(m)
+        expected *= sign
```

Comment:
I think this is wrong. The second argument of `slogdet` is already the log of the determinant. Multiplying it by the sign is going to do something nonsensical. 

The check for determinant being negative should be just `if sign < 0: `.

Reply:
Yep